bokeh/palettes.py

```diff
@@ -488,6 +511,8 @@
 
 def _cmap_func_generator(name, cmap):
     def func(n):
+        if n>256:
```

Comment:
Also I would say raise a `ValueError`. But I think this function needs a new name. This 256 restriction only applies to certain specific cases. Something extremely descriptive maybe: `_linear_cmap_from_256_colors`